package com.tencent.cos.xml.model.object;

import android.content.Context;
import android.net.Uri;
import com.tencent.cos.xml.CosXmlBaseService;
import com.tencent.cos.xml.common.ClientErrorCode;
import com.tencent.cos.xml.crypto.Headers;
import com.tencent.cos.xml.exception.CosXmlClientException;
import com.tencent.cos.xml.listener.CosXmlProgressListener;
import com.tencent.qcloud.core.http.x;
import java.io.File;
import java.io.InputStream;
import java.net.URL;
import java.util.Map;
import yb.b;
import yb.f;

/* JADX INFO: loaded from: classes7.dex */
public final class UploadPartRequest extends BaseMultipartUploadRequest implements TransferRequest {
    private byte[] data;
    private long fileContentLength;
    private long fileOffset;
    private InputStream inputStream;
    private boolean lastPart;
    private int partNumber;
    private CosXmlProgressListener progressListener;
    private String srcPath;
    private String uploadId;
    private Uri uri;
    private URL url;

    private UploadPartRequest(String str, String str2) {
        super(str, str2);
        this.fileOffset = -1L;
        this.fileContentLength = -1L;
        setNeedMD5(true);
    }

    @Override // com.tencent.cos.xml.model.object.ObjectRequest, com.tencent.cos.xml.model.CosXmlRequest
    public void checkParameters() throws CosXmlClientException {
        Context context;
        super.checkParameters();
        if (this.requestURL == null) {
            if (this.partNumber <= 0) {
                throw new CosXmlClientException(ClientErrorCode.INVALID_ARGUMENT.getCode(), "partNumber must be >= 1");
            }
            if (this.uploadId == null) {
                throw new CosXmlClientException(ClientErrorCode.INVALID_ARGUMENT.getCode(), "uploadID must not be null");
            }
        }
        String str = this.srcPath;
        if (str == null && this.data == null && this.inputStream == null && this.uri == null && this.url == null) {
            throw new CosXmlClientException(ClientErrorCode.INVALID_ARGUMENT.getCode(), "Data Source must not be null");
        }
        if (str != null && !new File(this.srcPath).exists()) {
            throw new CosXmlClientException(ClientErrorCode.INVALID_ARGUMENT.getCode(), "upload file does not exist");
        }
        Uri uri = this.uri;
        if (uri != null && (context = b.f241137a) != null && !f.a(uri, context.getContentResolver())) {
            throw new CosXmlClientException(ClientErrorCode.INVALID_ARGUMENT.getCode(), "upload file does not exist");
        }
    }

    public byte[] getData() {
        return this.data;
    }

    public long getFileContentLength() {
        return this.fileContentLength;
    }

    public long getFileLength() {
        if (this.data != null) {
            this.fileContentLength = r0.length;
        } else if (this.srcPath != null && this.fileContentLength == -1) {
            this.fileContentLength = new File(this.srcPath).length();
        }
        return this.fileContentLength;
    }

    public long getFileOffset() {
        return this.fileOffset;
    }

    @Override // com.tencent.cos.xml.model.CosXmlRequest
    public String getMethod() {
        return "PUT";
    }

    public int getPartNumber() {
        return this.partNumber;
    }

    public CosXmlProgressListener getProgressListener() {
        return this.progressListener;
    }

    @Override // com.tencent.cos.xml.model.CosXmlRequest
    public Map<String, String> getQueryString() {
        this.queryParameters.put("partNumber", String.valueOf(this.partNumber));
        this.queryParameters.put("uploadId", this.uploadId);
        return super.getQueryString();
    }

    @Override // com.tencent.cos.xml.model.CosXmlRequest
    public x getRequestBody() throws CosXmlClientException {
        if (this.srcPath != null) {
            return this.fileOffset != -1 ? x.e(getContentType(), new File(this.srcPath), this.fileOffset, this.fileContentLength) : x.d(getContentType(), new File(this.srcPath));
        }
        if (this.data != null) {
            return x.b(getContentType(), this.data);
        }
        if (this.inputStream != null) {
            return x.h(getContentType(), new File(CosXmlBaseService.appCachePath, String.valueOf(System.currentTimeMillis())), this.inputStream);
        }
        if (this.uri != null && b.f241137a != null) {
            return x.m(getContentType(), this.uri, b.f241137a, this.fileOffset, this.fileContentLength);
        }
        if (this.url != null) {
            return this.fileOffset != -1 ? x.o(getContentType(), this.url, this.fileOffset, this.fileContentLength) : x.n(getContentType(), this.url);
        }
        return null;
    }

    public String getSrcPath() {
        return this.srcPath;
    }

    public String getUploadId() {
        return this.uploadId;
    }

    public Uri getUri() {
        return this.uri;
    }

    public boolean isLastPart() {
        return this.lastPart;
    }

    public boolean isPriorityLow() {
        return this.priority == 1;
    }

    public void setData(byte[] bArr) {
        this.data = bArr;
    }

    public void setFileContentLength(long j10) {
        this.fileContentLength = j10;
    }

    public void setFileOffset(long j10) {
        this.fileOffset = j10;
    }

    public void setInputStream(InputStream inputStream) {
        this.inputStream = inputStream;
    }

    public void setLastPart(boolean z10) {
        this.lastPart = z10;
    }

    public void setPartNumber(int i10) {
        this.partNumber = i10;
    }

    public void setPriorityLow() {
        this.priority = 1;
    }

    public void setProgressListener(CosXmlProgressListener cosXmlProgressListener) {
        this.progressListener = cosXmlProgressListener;
    }

    public void setSrcPath(String str) {
        this.srcPath = str;
    }

    @Override // com.tencent.cos.xml.model.object.TransferRequest
    public void setTrafficLimit(long j10) {
        addHeader(Headers.COS_TRAFFIC_LIMIT, String.valueOf(j10));
    }

    public void setUploadId(String str) {
        this.uploadId = str;
    }

    public void setSrcPath(String str, long j10, long j11) {
        this.srcPath = str;
        this.fileOffset = j10;
        this.fileContentLength = j11;
    }

    public UploadPartRequest(String str, String str2, int i10, String str3, String str4) {
        this(str, str2);
        this.partNumber = i10;
        this.srcPath = str3;
        this.uploadId = str4;
        this.fileOffset = -1L;
        this.fileContentLength = -1L;
    }

    public UploadPartRequest(String str, String str2, int i10, Uri uri, String str3) {
        this(str, str2);
        this.partNumber = i10;
        this.uri = uri;
        this.uploadId = str3;
        this.fileOffset = -1L;
        this.fileContentLength = -1L;
    }

    public UploadPartRequest(String str, String str2, int i10, String str3, long j10, long j11, String str4) {
        this(str, str2);
        this.partNumber = i10;
        setSrcPath(str3, j10, j11);
        this.uploadId = str4;
    }

    public UploadPartRequest(String str, String str2, int i10, Uri uri, long j10, long j11, String str3) {
        this(str, str2);
        this.partNumber = i10;
        this.uri = uri;
        this.fileOffset = j10;
        this.fileContentLength = j11;
        this.uploadId = str3;
    }

    public UploadPartRequest(String str, String str2, int i10, byte[] bArr, String str3) {
        this(str, str2);
        this.partNumber = i10;
        this.data = bArr;
        this.uploadId = str3;
        this.fileOffset = -1L;
        this.fileContentLength = -1L;
    }

    public UploadPartRequest(String str, String str2, int i10, InputStream inputStream, String str3) throws CosXmlClientException {
        this(str, str2);
        this.partNumber = i10;
        this.inputStream = inputStream;
        this.uploadId = str3;
        this.fileOffset = -1L;
        this.fileContentLength = -1L;
    }

    public UploadPartRequest(String str, String str2, int i10, InputStream inputStream, long j10, String str3) throws CosXmlClientException {
        this(str, str2);
        this.partNumber = i10;
        this.inputStream = inputStream;
        this.uploadId = str3;
        this.fileOffset = 0L;
        this.fileContentLength = j10;
    }

    public UploadPartRequest(String str, String str2, int i10, URL url, String str3) {
        this(str, str2);
        this.partNumber = i10;
        this.url = url;
        this.uploadId = str3;
        this.fileOffset = -1L;
        this.fileContentLength = -1L;
        setNeedMD5(false);
    }

    public UploadPartRequest(String str, String str2, int i10, URL url, long j10, long j11, String str3) {
        this(str, str2);
        this.partNumber = i10;
        this.url = url;
        this.fileOffset = j10;
        this.fileContentLength = j11;
        this.uploadId = str3;
        setNeedMD5(false);
    }
}
